runtime/src/main/cpp/Types.cpp

```diff
@@ -97,8 +97,15 @@ KVector4f Kotlin_Vector4f_of(KFloat f0, KFloat f1, KFloat f2, KFloat f3) {
 	return {f0, f1, f2, f3};
 }
 
-KVector4i32 Kotlin_Vector4i32_of(KInt f0, KInt f1, KInt f2, KInt f3) {
-	return {f0, f1, f2, f3};
+/*
+ * In the current design all simd types are mapped internally to floating type, e.g. <4 x float>.
+ * However, some platforms (ex. arm32) have different calling convention for <4 x float> and <4 x i32>.
+ * To avoid illegal bitcast from/to function types the following function
+ * return type MUST be <4 x float> and explicit type cast is done on the variable type.
```

Comment:
I guess the same issue should be reproducible with `cinterop` when calling a function having int vector in signature.

Reply:
You're right. Looking at that.